### PR TITLE
Partner ServiceAreas

### PR DESCRIPTION
**Pull request by ivankocienski**
Migration, model, a few crude tests and some view code for service areas.

Still to do: accepting attributes for ServiceAreas in Partner create/update, verifications that Partners need either (or both) a service area or an address, a UI element so commissioners can select preferred option.

### CODE REVIEW

File: test/factories/service_areas.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :service_area do
+    partner_id { 1 }
+    neighbourhood_id { 1 }
```
Comment (kimadactyl) [inline test/factories/service_areas.rb:4]: think this only needs to be this to use the factory niceness (associations are implied)

```
partner
neighbourhood
```

Comment (ivankocienski) [reply test/factories/service_areas.rb:4]: ah i have yet to get to sorting out the factories. this is not a full PR

File: app/views/admin/partners/_form.html.erb [hunk 2]
```diff
@@ -22,6 +22,22 @@
   <br>
   <hr>
 
+  <h2>Service Area</h2>
+
+  <div class="partner__neighbourhoods">
+    <% @all_neighbourhoods.each do |neighbourhood| %>
+      <%= fields_for "partner[service_areas_attributes][#{neighbourhood.id}]" do |sna|%>
```
Comment (alexandria-gfsc) [inline app/views/admin/partners/_form.html.erb:29]: So I think? This is a place where simple_form's association dialogue would be appropriate

File: .ruby-gemset [hunk 3]
```diff
@@ -0,0 +1 @@
+gfsc-place-cal
```
Comment (alexandria-gfsc) [inline .ruby-gemset:1]: Is this likely to be applicable to other people's configurations? :)

File: test/models/service_area_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+describe ServiceArea do
+  # it "does a thing" do
+  #   value(1+1).must_equal 2
+  # end
+end
```
Comment (alexandria-gfsc) [inline test/models/service_area_test.rb:7]: gotta remove this file :)

File: .ruby-gemset (hunk 3 above)
Comment (kimadactyl) [reply .ruby-gemset:1]: yeah let's not specify environment, we can add to `.gitignore` if needs be

File: app/views/admin/partners/_form.html.erb (hunk 2 above)
Comment (kimadactyl) [reply app/views/admin/partners/_form.html.erb:29]: `|sna| %>` missing space here is killing me lmao

File: .ruby-gemset (hunk 3 above)
Comment (ivankocienski) [reply .ruby-gemset:1]: rbenv has a plugin allowing you a per-project set of gems (https://github.com/jf/rbenv-gemset). I find it keeps my dev environment lightweight and allows me to have multiple projects with different versions of gems at the same time. You can ignore and it won't interfere with anything, but it is super useful

Comment (kimadactyl) [reply .ruby-gemset:1]: oh i hear you and this sounds good. just as project maintainers we don't really want to be in the business of telling people what their local development environment should be. dotfiles have a way of piling up you know? i think if you are really keen on putting this in the codebase then we should be adding stuff to the README etc about our suggested environment. maybe its fine for now tho as long as we have a note to do this later. altho pls remove space :) i.e. `gfsc-placecal`

Comment (ivankocienski) [reply .ruby-gemset:1]: I see your point. dotfiles Is there a formal process for suggesting technologies? Like if we need to set up a background job system and needed to add redis to our dev/prod machines.

I checked the README and it doesn't mention anything about requirements at all. I could put something in right now to cover rbenv, rbenv-ruby ?

Comment (kimadactyl) [reply .ruby-gemset:1]: yeah this is a great example of something that sounds great _one day_ but we truly do not need any time soon :)

you're welcome to file an issue and label it "code smell" or "tech debt" tho!

**Comment by alexandria-gfsc** [comment]
LGTM :)

**Comment by alexandria-gfsc** [comment]
idk if you would prefer to squash/merge or just merge, up to you :+1:

**Pull request merged**